lib/bolt/error.rb

```diff
@@ -42,4 +42,12 @@ def initialize(resultset, action, object)
       @error_code = 2
     end
   end
+
+  class PlanFailure < Error
+    def initialize(*args)
+      super(*args)
+      # CODEREVIEW: what should this be?
+      @error_code = 3
```

Comment:
I think the message probably matters more than the numeric code? We could also have 1xx be task failures, and 2xx be plan failures. But ultimately I think the message is more important.

Reply:
this eventually become the exit code so we probably don't want to stake out that much space. 

I think the question is should the exitcode be
1. we use this for general bolt failures
2. used for "runfailure" when a bolt execution failed automatically because a task/script/command/upload failed on some targets.
3. Some new number to differentiate explicit plan failures from run_* automatic failures.

Reply:
I figured plan failure would be like other execution failures.  Otherwise 3 seems fine (and we should document that somewhere).